editor/src/components/canvas/canvas-wrapper-component.tsx

```diff
@@ -93,11 +119,296 @@ export const CanvasWrapperComponent = React.memo(() => {
   const isNavigatorOverCanvas = useEditorState(
     Substores.restOfEditor,
     (store) => !store.editor.navigator.minimised,
-    'ErrorOverlayComponent isOverlappingWithNavigator',
+    'CanvasWrapperComponent isOverlappingWithNavigator',
   )
 
-  const navigatorWidth = usePubSubAtomReadOnly(NavigatorWidthAtom, AlwaysTrue)
   const updateCanvasSize = usePubSubAtomWriteOnly(CanvasSizeAtom)
+  const navigatorWidth = usePubSubAtomReadOnly(NavigatorWidthAtom, AlwaysTrue)
+
+  const actualNavigatorWidth = React.useMemo(() => {
+    if (!isNavigatorOverCanvas) {
+      return 0
+    }
+    return navigatorWidth
+  }, [navigatorWidth, isNavigatorOverCanvas])
+
+  const ref = React.useRef<HTMLDivElement | null>(null)
+
+  const [mousePoint, setMousePoint] = React.useState<CanvasPoint | null>(null)
+  const [selectionAreaStart, setSelectionAreaStart] = React.useState<CanvasPoint | null>(null)
+  const [elementsUnderSelectionArea, setElementsUnderSelectionArea] = React.useState<ElementPath[]>(
+    [],
+  )
+
+  const mode = useEditorState(
+    Substores.restOfEditor,
+    (store) => store.editor.mode,
+    'CanvasWrapperComponent mode',
+  )
+  const metadata = useEditorState(
+    Substores.metadata,
+    (store) => store.editor.jsxMetadata,
+    'CanvasWrapperComponent metadata',
+  )
```

Comment:
I'm not sure how big a problem this is, but this will cause a re-render on each metadata change, I'd check the impact with the re-rendering highlights on just to make sure